src/wp-includes/blocks/index.php

```diff
@@ -40,26 +39,62 @@ function register_core_block_style_handles() {
 		'editorStyle' => 'editor',
 	);
 
+	static $core_blocks_meta;
+	if ( ! $core_blocks_meta ) {
+		$core_blocks_meta = require $includes_path . 'blocks/blocks-json.php';
+	}
+
+	$files          = false;
+	$transient_name = 'wp_core_block_css_files';
 	/*
 	 * Ignore transient cache when the development mode is set to 'core'. Why? To avoid interfering with
 	 * the core developer's workflow.
 	 */
 	if ( ! wp_is_development_mode( 'core' ) ) {
-		$transient_name = 'wp_core_block_css_files';
-		$files          = get_transient( $transient_name );
-		if ( ! $files ) {
-			$files = glob( wp_normalize_path( __DIR__ . '/**/**.css' ) );
-			set_transient( $transient_name, $files );
+		$cached_files = get_transient( $transient_name );
+
+		/*
+		 * Check the validity of cached values.
+		 *  - Should match the current WordPress version
+		 *  - Should use relative paths for the files
+		 */
+		if ( is_array( $cached_files ) ) {
+			if ( ! isset( $cached_files['version'] ) || $cached_files['version'] !== $wp_version ) {
+				$files = false;
+			} elseif ( ! isset( $cached_files['files'] ) || empty( $cached_files['files'] ) || ! str_starts_with( reset( $cached_files['files'] ), 'blocks/' ) ) {
+				$files = false;
```

Comment:
I think this can now be removed? We only had this here to check for older cache values from pre-6.4 where the full paths were being cached, but that's now already handled by the version check. So I think this check no longer serves a purpose now, given we _always_ cache relative paths.

We should either remove it completely, or for defensive coding keep only the `isset` part.
```suggestion
			} elseif ( ! isset( $cached_files['files'] ) ) {
				$files = false;
```

Reply:
I went with your other suggestion which clean the code nicely and is more concise.